src/Build/Construction/ProjectSdkElement.cs

```diff
@@ -35,45 +52,46 @@ internal ProjectSdkElement(ProjectElementContainerLink link)
         /// </summary>
         private ProjectSdkElement(XmlElementWithLocation xmlElement, ProjectRootElement containingProject)
             : base(xmlElement, null, containingProject)
-        { }
+        {
+        }
 
         /// <summary>
         /// Gets or sets the name of the SDK.
         /// </summary>
         public string Name
         {
-            get => GetAttributeValue(XMakeAttributes.sdkName);
+            get => GetAttributeValue(XMakeAttributes.sdkName, true);
             set
             {
                 ErrorUtilities.VerifyThrowArgumentLength(value, XMakeAttributes.sdkName);
-                SetOrRemoveAttribute(XMakeAttributes.sdkName, value, $"Set SDK Name to {value}", XMakeAttributes.sdkName);
```

Comment:
Fix reason messages here and in the following 2 properties (`param` was not used, inconsistent with `ProjectImportElement`).